cluster-autoscaler/processors/pods/pod_list_processor.go

```diff
@@ -21,12 +21,11 @@ import (
 	"k8s.io/autoscaler/cluster-autoscaler/context"
 )
 
-// PodListProcessor processes lists of unschedulable and scheduled pods before scaling of the cluster.
+// PodListProcessor processes lists of unschedulable pods.
 type PodListProcessor interface {
-	Process(context *context.AutoscalingContext,
-		unschedulablePods []*apiv1.Pod, allScheduledPods []*apiv1.Pod,
-		allNodes []*apiv1.Node, readyNodes []*apiv1.Node,
-		upcomingNodes []*apiv1.Node) ([]*apiv1.Pod, []*apiv1.Pod, error)
+	Process(
+		context *context.AutoscalingContext,
+		unschedulablePods []*apiv1.Pod) ([]*apiv1.Pod, error)
```

Comment:
What about pre-existing PLPs that mutated scheduled pods list? Are they supposed to mutate snapshot instead?

Reply:
Ok, I figured it out after reading filterOutSchedulable. Resolving.

Reply:
Yeah - that is current approach.